src/main/scala/chisel3/util/SRAM.scala

```diff
@@ -79,14 +79,18 @@ class MemoryReadWritePort[T <: Data](tpe: T, addrWidth: Int, masked: Boolean) ex
   * @param numReadPorts The number of read ports
   * @param numWritePorts The number of write ports
   * @param numReadwritePorts The number of read/write ports
+  * @param masked Whether the memory is write masked
+  * @param underlying Points to SRAM that instantiates this interface; None by
+  *                   default if this was instantiated directly
   */
 class SRAMInterface[T <: Data](
-  memSize:           BigInt,
-  tpe:               T,
-  numReadPorts:      Int,
-  numWritePorts:     Int,
-  numReadwritePorts: Int,
-  masked:            Boolean = false)
+  memSize:                          BigInt,
+  tpe:                              T,
+  numReadPorts:                     Int,
+  numWritePorts:                    Int,
+  numReadwritePorts:                Int,
+  masked:                           Boolean = false,
+  private[chisel3] val _underlying: Option[HasTarget] = None)
```

Comment:
This isn't doing quite what you want it to, the constructor is public and thus users still have the ability to pass the argument. This really needs to be a `private[chisel3] var _underlying` down by `def` underlying (and then mutably set after running the constructor down below)